py/desisim/obs.py

```diff
@@ -29,7 +29,7 @@
 from .simexp import simulate_spectra
 
 def new_exposure(program, nspec=5000, night=None, expid=None, tileid=None,
-                 seed=None, obsconditions=None, testslit=False, exptime=None,
+                 seed=None, obsconditions=None, specify_targets=None, testslit=False, exptime=None,
```

Comment:
This implementation with `specify_targets` will work for mono-target programs like "BGS" but not multi-target programs like "BRIGHT" or "DARK" where you may want different selections for different types of templates.  I think we need something a bit messier but more general:

`new_exposure` has separate options for each target class with a default as a blank dict, e.g.
```
def new_exposure(..., bgs_kwargs=dict(), elg_kwargs=dict(), lrg_kwargs=dict(), ...)
```
and then down in targets.select_targets, each individual class would use it individual options,
```
simflux, wave1, meta1 = elg.make_templates(nmodel=nobj, seed=seed, **elg_kwargs)
simflux, wave1, meta1 = lrg.make_templates(nmodel=nobj, seed=seed, **lrg_kwargs)
...
```
by making the default a `dict()` instead of None, you can also avoid the if/else logic for whether any keyword args are specified or not.